src/Handshake/ClientNegotiator.php

```diff
@@ -25,6 +25,22 @@ function __construct() {
           , 'Sec-WebSocket-Version' => $this->getVersion()
           , 'User-Agent'            => "Ratchet"
         ]);
+
+        if ($perMessageDeflateOptions === null) {
+            $perMessageDeflateOptions = PermessageDeflateOptions::createDisabled();
+        }
+
+        // https://bugs.php.net/bug.php?id=73373
+        // https://bugs.php.net/bug.php?id=74240 - need >=7.1.4 or >=7.0.18
+        $supported = PermessageDeflateOptions::permessageDeflateSupported();
+        if (!$supported) {
```

Comment:
If the permessage deflate is disabled, why would you check whether it is supported?.

It would make sense to make this check conditional on whether it is enabled or not. Currently you're simply overwriting the variable, even if it is already disabled.

```php
if (
    $perMessageDeflateOptions->isEnabled() &&
    !PermessageDeflateOptions::permessageDeflateSupported()
) {
    trigger_error('permessage-deflate is being disabled because it is not support by your PHP version.', E_USER_NOTICE);
}
```